framework/src/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -73,7 +73,28 @@ public static void initializeTypeArgs(AnnotatedDeclaredType declaredType) {
                 TypeMirror upperBound =
                         ((TypeVariable) typeElement.getTypeParameters().get(i).asType())
                                 .getUpperBound();
-                javaTypeArg = declaredType.atypeFactory.types.getWildcardType(upperBound, null);
+                switch (upperBound.getKind()) {
+                    case ARRAY:
+                    case DECLARED:
+                    case TYPEVAR:
+                        javaTypeArg =
+                                declaredType.atypeFactory.types.getWildcardType(upperBound, null);
+                        break;
+                    case INTERSECTION:
+                        TypeMirror object =
+                                TypesUtils.getObjectTypeMirror(
+                                        declaredType.atypeFactory.processingEnv);
+                        javaTypeArg = declaredType.atypeFactory.types.getWildcardType(object, null);
+                        break;
+                    default:
+                        assert false;
```

Comment:
Instead of the `assert` could you do an `errorAbort` and remove the next two assignments? I'm not quite sure why these statements are here.